roofit/roofitcore/src/RooMinimizer.cxx

```diff
@@ -96,43 +97,81 @@ void RooMinimizer::cleanup()
 /// for HESSE and MINOS error analysis is taken from the defaultErrorLevel()
 /// value of the input function.
 
-RooMinimizer::RooMinimizer(RooAbsReal &function, FcnMode fcnMode) : _fcnMode(fcnMode)
+/// Construcor that accepts all configuration in struct with RooAbsReal likelihood
+RooMinimizer::RooMinimizer(RooAbsReal &function, Config &cfg) : _cfg(cfg)
 {
+   // Call init functions
    initMinimizerFirstPart();
+   if (_cfg.parallelGradient || _cfg.parallelLikelihood) {
+      RooFit::TestStatistics::RooRealL *nll_real = dynamic_cast<RooFit::TestStatistics::RooRealL *>(&function);
```

Comment:
```suggestion
      auto nll_real = dynamic_cast<RooFit::TestStatistics::RooRealL *>(&function);
```
If the type name is very long and it's clear what the type is from the right hand side already (as is there case here because of the case), it's good practice to use `auto` to make the code more readable.